### DISCUSSION TITLE
Reset Interval

### DISCUSSION THREAD

**Discussion by usernotnull** [alpine-collective/alpine-magic-helpers#58] [Ideas] [answered]
I'm running an $interval, and I had to reset its value when the user clicks.
Is there a way to deal with it at this stage?
If not, maybe a $intervalReset would be useful?
And maybe an $intervalPause(2000) to delay the next iteration by 2000ms?

Those helpers would save me a lot of code (adding more variables and timers and booleans to achieve the above).

**Comment by KevinBatdorf** [comment, marked as answer]
You can clear the timer on click with something like `<button @click="autoIntervalTest = false">` and you could abstract that to a function that will set it to true after 2 seconds.

Internally, there's a watcher for that property

```js
this.$watch('autoIntervalTest', test => {
  if (test) {
    this.autoIntervalLoop = loop

    forceInterval ? this.autoIntervalLoop() : requestAnimationFrame(this.autoIntervalLoop)
  } else {
    clearTimeout(this.autoIntervalLoop)

    this.autoIntervalLoop = null
  }
})

**Comment by usernotnull** [comment]
Oh lovely, that was what I needed actually.
The "it's truthiness" part in the docs wasn't clear for me but now I understood.
Thanks.

**Reply by KevinBatdorf** [reply]
Hmm. Yeah maybe I can make that a bit more direct in the docs.
